frontend/src/app/plans/edit/[id]/page.client.tsx

```diff
@@ -208,115 +113,93 @@ export function CreateNewPlanPage({
     },
   });
 
-  const handleUpdateLocalPlan = async () => {
+  //- Plan state functions -//
+  const createOnlinePlan = async () => {
     firstTime.current = false;
+    await handleCreateOnlinePlan(plan, setOfflineAlert);
+  };
 
-    if (!onlinePlan) {
-      return false;
-    }
-
-    let updatedRegistrations: typeof plan.registrations = [];
-    let updatedCourses: typeof plan.courses = [];
+  const syncPlan = async () => {
+    await handleSyncPlan(plan, refetchOnlinePlan, setSyncing);
+  };
 
-    for (const registration of onlinePlan.registrations) {
-      try {
-        const courses = await coursesFn.mutateAsync(registration.id);
-        const extendedCourses: ExtendedCourse[] = courses
-          .map((c) => {
-            const groups: ExtendedGroup[] = c.groups.map((g) => ({
-              groupId: g.group + c.id + g.type,
-              groupNumber: g.group.toString(),
-              groupOnlineId: g.id,
-              courseId: c.id,
-              courseName: c.name,
-              isChecked:
-                onlinePlan.courses
-                  .find((oc) => oc.id === c.id)
-                  ?.groups.some((og) => og.id === g.id) ?? false,
-              courseType: g.type,
-              day: g.day,
-              lecturer: g.lecturer,
-              registrationId: c.registrationId,
-              week: g.week.replace("-", "") as "" | "TN" | "TP",
-              endTime: g.endTime.split(":").slice(0, 2).join(":"),
-              startTime: g.startTime.split(":").slice(0, 2).join(":"),
-            }));
-            return {
-              id: c.id,
-              name: c.name,
-              isChecked: onlinePlan.courses.some((oc) => oc.id === c.id),
-              registrationId: c.registrationId,
-              type: c.groups.at(0)?.type ?? ("" as LessonType),
-              groups,
-            };
-          })
-          .sort((a, b) => a.name.localeCompare(b.name));
+  const updateLocalPlan = async () => {
+    firstTime.current = false;
+    await handleUpdateLocalPlan(onlinePlan, plan, coursesFn);
+  };
 
-        // List update logic:
-        // Add unique registrations to the updatedRegistrations array
-        // r - current registration
-        // i - current index
-        // a - array of registrations
-        updatedRegistrations = [...updatedRegistrations, registration].filter(
-          (r, i, a) => a.findIndex((t) => t.id === r.id) === i,
-        );
```

Comment:
wiem ze juz wywalone, ale w takim przypadku nie lepiej od razu nazwać zmienne r -> currentRegistration, zamiast pisać komentarz co to oznacza?

Reply:
no lepiej lepiej, droczylem sie z bartkiem